### PR TITLE
Concourse pipeline suggestions

### PR DESCRIPTION
**Pull request by krishicks**
See the commits for explanation for each one.

Note that I didn't actually run this pipeline with these edits; this PR should be taken as informational only.

You might consider using [serial groups](http://concourse.ci/single-page.html#serial_groups) rather than `serial: true`, as the latter only prevents multiple instances of the same job from running, whereas the former prevents multiple jobs with the same serial group tag from running at the same time. For example, the two jobs that do a `put: repo` could be in a serial group called "repo"; then they wouldn't both be able to run at the same time, reducing the possibility of contention on the `repo` resource.

`public: false` isn't necessary as it's the default, but feel free to leave it if you like to be explicit. Same goes for `trigger: false`.

Also, check the messages in each task; I think some are wrong. For example, the stage-deploy task [says](https://github.com/spring-cloud/spring-cloud-pipelines/blob/master/concourse/tasks/stage-deploy/task.sh#L16) "Deploying the built application on _test_ environment". Should be _stage_.

### CODE REVIEW

**Comment by krishicks** [comment]
Something else I'd add is that whenever shipping Concourse pipelines as a product it makes sense to also add some tests around the pipelines. I wrote a bunch of Ginkgo tests for pcf-pipelines that I then extracted into a tool, [testpipe](https://github.com/krishicks/testpipe), which can be used to test certain aspects of Concourse pipelines.

One of the tests that `testpipe` has is to make sure that the params you specify in the pipeline.yml match exactly those that are expected by the task.yml; in the spring-cloud-pipelines tasks you don't currently specify params so this wouldn't work. But we do it on pcf-pipelines because the tasks are shared by many pipelines and we want to make sure that whenever someone changes a task the pipeline is also changed accordingly.

File: concourse/pipeline.yml [hunk 1]
```diff
@@ -162,8 +172,10 @@ jobs:
       - test-rollback-smoke
       trigger: false
   - task: stage-deploy
-    file: tools/concourse/tasks/stage-deploy/task.yml
+    file: tools/concourse/task.yml
     params:
+      MESSAGE: "Deploying the built application on test environment"
```
Comment (marcingrzejszczak) [inline concourse/pipeline.yml]: Yup it should be stage

File: concourse/task.sh [hunk 2]
```diff
@@ -13,7 +13,7 @@ echo "Version resource folder is [${VERSION_RESOURCE}]"
 
 source ${ROOT_FOLDER}/${TOOLS_RESOURCE}/concourse/tasks/pipeline.sh
 
-echo "Completing the switch over"
+echo $MESSAGE
```
Comment (marcingrzejszczak) [inline concourse/task.sh]: Makes perfect sense

File: concourse/task.sh [hunk 3]
```diff
@@ -13,7 +13,7 @@ echo "Version resource folder is [${VERSION_RESOURCE}]"
...
 cd ${ROOT_FOLDER}/${REPO_RESOURCE}
 
-. ${SCRIPTS_OUTPUT_FOLDER}/prod_complete.sh
+. ${SCRIPTS_OUTPUT_FOLDER}/${SCRIPT_TO_RUN}
```
Comment (marcingrzejszczak) [inline concourse/task.sh]: This change also makes perfect sense. Thanks for catching it.

**Comment by marcingrzejszczak** [comment]
So if I understand correctly this PR does some cleaning of scripts + removes all custom scripts to reuse a single one with 2 params - script to be executed and a message, right?

As for testing - I'd be more than happy to add some additional ones. As you can see in the `common` module we're already using a lot of `bats` tests. I was thinking about adding the same thing for Concourse but actually in Concourse we're just delegating work to a script so maybe it doesn't make much sense...  We could definitely use it.

You're saying that the params set in `pipeline.yml` don't match those in `task.yml`. We can, of course, change the code to match these if you think that it makes sense.

**Comment by krishicks** [comment]
Sorry, I didn't get notifications of your message.

> So if I understand correctly this PR does some cleaning of scripts + removes all custom scripts to reuse a single one with 2 params - script to be executed and a message, right?

It removes most of the custom scripts because they mostly all did the same thing; the two that weren't changed have extra tagging steps that could have been split into another task ("tag-repo" or something).

> As for testing - I'd be more than happy to add some additional ones. As you can see in the common module we're already using a lot of bats tests. I was thinking about adding the same thing for Concourse but actually in Concourse we're just delegating work to a script so maybe it doesn't make much sense... We could definitely use it.

Try running testpipe against it for a start.

> You're saying that the params set in pipeline.yml don't match those in task.yml. We can, of course, change the code to match these if you think that it makes sense.

It's good to know that the way Concourse does it is by merging in values from the pipeline task declaration into the task's config, so it isn't strictly necessary to do this. However, the value in making them match has been pretty high for us since we have so many shared tasks; this repo after my change also has a lot of shared tasks so it makes sense to do a similar thing to make sure you don't forget to add/remove/replace a param at some point in the future. It just shortens the feedback cycle if you add testpipe to your CI.

**Pull request merged**